### DISCUSSION TITLE
Configure VPN connection for MySQL

### DISCUSSION THREAD

**Discussion by divdax** [ddev/ddev#4556] [Q&A] [answered]
Is it possible to configure ddev so any outgoing MySQL connections are made over my customers vpn so i don't have to configure the vpn on every host machine to access external database?

**Comment by rfay** [comment]
Here's how I responded in Discord:

> I would imagine it's possible if the customer has a linux client for the VPN, and you can figure out a way to initialize the VPN in the container without breaking docker networking, and you find a good way to manage the VPN secrets. I don't think I recommend it. Will the customer not let you use a copy of the DB in your local dev environment? That's the normal and easy way...

Please don't ask a bunch of places; but when you sort this out you can put the solution here.

**Comment by rachellawson** [comment]
I can see many use-cases for a vpn client inside the ddev network. Yes, it would be normal to have a copy of the drupal database locally but there may be other services that the site needs to talk to (Drupal being so annoyingly connectable).

I’m specifically thinking our environment might want to talk to an external mongoldb and solr service, both of which contain information that drupal must consult, even if it doesn’t update it.

So, I’m going to spend some time on this. A report on any investigations done so far, @divdax , would be appreciated!

**Reply by rfay** [reply, marked as answer]
In most cases, VPNs will work fine from the docker network to the outside world. This doesn't really have anything to do with DDEV.

There are a couple of exceptions:

* VPN doesn't share the host internet connection with the Docker network. This is configuration and should be solvable. It can also be solved by using a proxy tcp to the host side.
* VPN does packet inspection and changes the certificates required for the internet. This one can be fixed with https://stackoverflow.com/a/71595428/215713 (see other answers there as well). In your case though, you'd install in .ddev/db-build/Dockerfile instead of .ddev/web-build
